### PR TITLE
[Intel MKL] Klocwork fix

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**

### CODE REVIEW

**Comment by mahmoud-abuzaina** [comment]
Can we cherry-pick this into r2.1 as this is required to release TF 2.1? Here is the PR against r2.1 #35391

**Review by alextp — CHANGES_REQUESTED** [review]
Is there a unit test which fails before this change?

**Comment by mahmoud-abuzaina** [comment]
@alextp No unit test is failing. Klocwork is a tool that scans code for security and reliability issues. Intel cannot release code if it does not pass this scan. The change in this PR has no functional impact. It is needed just to make the tool scan pass which requires every member variable to be initialized in the constructor.

File: tensorflow/core/common_runtime/eager/mkl_eager_op_rewrite.cc [hunk 1]
```diff
@@ -88,6 +88,7 @@ REGISTER_REWRITE(EagerOpRewriteRegistry::PRE_EXECUTION, MklEagerOpRewrite);
 // Constructor
 MklEagerOpRewrite::MklEagerOpRewrite(string name, string file, string line)
     : EagerOpRewrite(name, file, line) {
+  registered_kernels_map_ = std::unordered_map<std::string, bool>();
```
Comment (alextp) [inline tensorflow/core/common_runtime/eager/mkl_eager_op_rewrite.cc]: Then isn't it better to initialize it in the : line? Line, MklEagerOpRewrite::MklEagerOpRewrite(string name, string file, string line)
    : EagerOpRewrite(name, file, line), registered_kernel_map_() { ...

?

Comment (mahmoud-abuzaina) [reply tensorflow/core/common_runtime/eager/mkl_eager_op_rewrite.cc]: That is right. I will do that.

**Review by alextp — APPROVED** [review]

**Pull request merged**